### PR TITLE
Add delete-namespace option when uninstall

### PR DESCRIPTION
**Pull request by ThibaultDelaune-pro**
Usually, when I uninstall an Helm release, I delete the namespace afterwards. This option would permit that in a single task.

### CODE REVIEW

**Comment by Typositoire** [comment]
Concourse: Version bump found.

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.22.1
\ No newline at end of file
+1.22.2
```
Comment (Typositoire) [inline VERSION:1]: Please make this a minor bump since it introduces a behavior change.

File: assets/out [hunk 2]
```diff
@@ -282,6 +283,12 @@ if [ "$uninstall" = true ]; then
   helm_uninstall
   result="$(jq -n "{version:{release:\"$release\", uninstalled: \"true\"}, metadata: [{name: \"release\", value: \"$release\"}]}")"
   echo "$result" | jq -s add  >&3
+
+  if [ "$delete_namespace" = true ] && [ -n "$namespace" ] && [ $debug = false ]; then
+    echo "Deleting the namespace $namespace..."
+    logfile="/tmp/log"
```
Comment (Typositoire) [inline assets/out:289]: I think you're missing a `mkdir -p /tmp` after this line.

File: assets/out [hunk 3]
```diff
@@ -282,6 +283,12 @@ if [ "$uninstall" = true ]; then
...
+    kubectl delete namespaces --ignore-not-found=true "$namespace" | tee "$logfile"
```
Comment (Typositoire) [inline assets/out:290]: In which case would we ask for delete and the namespace would not exist? I don't think the resource should prevent user error if the namespace it is trying to delete doesn't exist.

**Review by Typositoire — CHANGES_REQUESTED** [review]

File: assets/out (hunk 3 above)
Comment (Berber31) [reply assets/out:290]: yes, but `--ignore-not-found=true` is bulletproof

File: assets/out (hunk 2 above)
Comment (Berber31) [reply assets/out:289]: i think already done in [helm_uninstall function](https://github.com/Typositoire/concourse-helm3-resource/blob/494c1af10610c5c4e960723450b1aeffbd98ffc5/assets/out#L253)

File: assets/out (hunk 3 above)
Comment (Typositoire) [reply assets/out:290]: I think using https://concourse-ci.org/try-step.html is more flexible than FORCING the resource to ignore any kind of error. Maybe people wants to error out if the namespace doesn't exist.

File: assets/out (hunk 2 above)
Comment (Typositoire) [reply assets/out:289]: Oh wow ok I didn't realize this was at that place. Then that part should be BEFORE the `echo "$result"`. echo should be the last thing that happens.

File: VERSION (hunk 1 above)
Comment (ThibaultDelaune-pro) [reply VERSION]: Done!

File: assets/out (hunk 2 above)
Comment (ThibaultDelaune-pro) [reply assets/out:289]: Fixed!

File: assets/out (hunk 3 above)
Comment (ThibaultDelaune-pro) [reply assets/out]: > yes, but `--ignore-not-found=true` is bulletproof

This was the idea 😉

Comment (ThibaultDelaune-pro) [reply assets/out]: I removed the param

**Comment by Typositoire** [comment]
Concourse: Version bump found.

**Review by Typositoire — CHANGES_REQUESTED** [review]
Fix the version then we're good to go!

File: VERSION [hunk 4]
```diff
@@ -1 +1 @@
-1.22.1
\ No newline at end of file
+1.23.1
```
[the remark this answers is not in the document]
Comment (ThibaultDelaune-pro) [reply VERSION]: Done!
'hope this will do... 😋

**Comment by Typositoire** [comment]
Concourse: Version bump found.

**Review by Typositoire — APPROVED** [review]

**Pull request merged**